app/src/main/java/com/infomaniak/drive/ui/home/RootFilesFragment.kt

```diff
@@ -82,46 +89,96 @@ class RootFilesFragment : Fragment() {
         )
 
         setupRootPendingFilesIndicator(mainViewModel.pendingUploadsCount, rootFilesUploadFileInProgressView)
+
+        if (fileListViewModel.justLaunched.not()) {
+            uiSettings.lastVisitedRootFileTreeCategory = null
+        }
+        if (fileListViewModel.justLaunched) lifecycleScope.launch {
+            fileListViewModel.justLaunched = false
+            when (uiSettings.lastVisitedRootFileTreeCategory) {
+                CommonFolders -> {
+                    folderToOpenSet.join()
+                    commonFolderToOpen?.let { (id, name) ->
+                        safeNavigate(RootFilesFragmentDirections.actionFilesFragmentToFileListFragment(folderId = id, folderName = name))
+                    }
+                }
+                PersonalFolder -> {
+                    folderToOpenSet.join()
```

Comment:
I dont really get what you were meaning with this name. 
What is something along the lines of 'setFolderToOpenJob` like it's the Job made for setting the folder to open ?

Reply:
The `folderToOpenSet` property is a `CompletableJob`, a signal to await for the `commonFolderToOpen` and `personalFolderToOpen` properties to be set, since those are required to navigate to the "Common folder" and "Personal folder" routes.